pkg/metrics/metricbuilder/metricbuilder.go

```diff
@@ -0,0 +1,64 @@
+package metricbuilder
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// MetricBuilder is the basic metric type that must be used by any potential metric added to
+// the system. Any metric added must have the following functions which would be used in the
+// workflow.
+type MetricBuilder struct {
+	labels         []string
+	labelKeyValues map[string]string
+	desc           string
+	name           string
+	value          float64
+	buckets        []float64
+	metricType     string
```

Comment:
rather than `string ` I think this should be a type, simulating an enum using this pattern: https://github.com/openshift/installer/blob/master/pkg/types/azure/platform.go#L69-L83

Only values corresponding to the prometheus metrics type should be allowed here, so `histogram`, `counter`, `gauge`